sdk/tables/data-tables/src/utils/errorHelpers.ts

```diff
@@ -0,0 +1,62 @@
+import { OperationOptions, OperationRequest } from "@azure/core-client";
+import { PipelineResponse, RestError } from "@azure/core-rest-pipeline";
+import { SpanStatusCode } from "@azure/core-tracing";
+import { AzureLogger } from "@azure/logger";
+import { TableServiceError } from "../generated";
+
+export type TableServiceErrorResponse = PipelineResponse & {
+  /**
+   * The parsed HTTP response headers.
+   */
+  parsedHeaders?: Record<string, unknown>;
+  /**
+   * The response body as parsed JSON or XML.
+   */
+  parsedBody: TableServiceError;
+  /**
+   * The request that generated the response.
+   */
+  request: OperationRequest;
+};
+
+export function handleTableAlreadyExist(
+  error: unknown,
+  options: OperationOptions & { span?: any; logger?: AzureLogger } = {}
+) {
+  const responseError = getErrorResponse(error);
+  if (
+    responseError &&
+    responseError.status === 409 &&
+    responseError.parsedBody.odataError?.code === "TableAlreadyExists"
+  ) {
+    options.logger?.info("Table Already Exists");
+    options.onResponse && options.onResponse(responseError, {});
```

Comment:
This actually surprised me, but yeah digging into it we never call onResponse when a RestError gets thrown:

https://github.com/Azure/azure-sdk-for-js/blob/c211cb62cc61ccdb5cef2fa7254ed830b24ab035/sdk/core/core-client/src/serviceClient.ts#L194

I wonder if this is the right behavior though. Could you file an issue and tag myself, Brian, and Jeremy? I think we should discuss possibly changing that, at least for the cases where we get back an HTTP response

Reply:
Makes sense filed #18949 for discussing